### PR TITLE
Set custom user agent

### PR DESCRIPTION
**Pull request by guyilin-amazon**
*Issue #, if available:*
#196 
*Description of changes:*
Set customized user agent, now the user agent will be like `aws-sdk-rust/0.1.0 api/QLDB Driver for Rust v0.1.0/QLDB Shell for Rust v2.0.0 os/macos lang/rust/1.55.0`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/awssdk_driver.rs [hunk 1]
```diff
@@ -38,6 +39,8 @@ impl<C> QldbSessionSdk<C> {
         }
     }
 }
+const DRIVER_VERSION: &'static str = "QLDB Driver for Rust v0.1.0";
+const SHELL_VERSION: &'static str = "QLDB Shell for Rust v2.0.0";
```
Comment (marcbowes) [inline src/awssdk_driver.rs:43]: Please can you use the same mechanism as was previously in use, where we don't need to manually keep this up to date?

**Review by marcbowes — CHANGES_REQUESTED** [review]

File: src/awssdk_driver.rs (hunk 1 above)
Comment (marcbowes) [reply src/awssdk_driver.rs:43]: https://github.com/awslabs/amazon-qldb-shell/blob/b18304949f98f26b29428400275ddb86f0c68f5e/src/rusoto_driver.rs#L96-L101

**Comment by marcbowes** [comment]
Also, since it's hard to write a test for this, can you show evidence that this works? For example, you can enable file logging in the config and then look at a sample request. Alternatively, you can look at a request on the service side and check the metrics work.

File: Cargo.toml [hunk 2]
```diff
@@ -10,12 +10,13 @@ edition = "2018"
 amazon-qldb-driver = { git = "https://github.com/awslabs/amazon-qldb-driver-rust", package = "amazon-qldb-driver", branch = "main" }
```
Comment (marcbowes) [inline Cargo.toml:10]: We need to get the sdk updated in the driver too.

**Comment by rcoh** [comment]
FYI: this can be tested quite easily by using `capture_request()`: https://github.com/awslabs/aws-sdk-rust/blob/main/sdk/aws-smithy-client/src/test_connection.rs#L75-L106

(note: `test-util` must be enabled)

**Pull request merged**